### PR TITLE
[AC-7965] Fix display of time in Office Hour reservation emails

### PR DESCRIPTION
**Pull request by jonkiparsky**
https://masschallenge.atlassian.net/browse/AC-7965

To verify locally:
Create an office hour, ideally with a Location not in your local timezone
Use the reserve_office_hour endpoint localhost:8000/api/v1/reserve_office_hour  to reserve that office hour
Check the output in your terminal: should see two emails displayed, both of them having the correct start time for the office hour, displayed in the timezone set on that office hour.

### CODE REVIEW

File: web/impact/impact/v1/views/reserve_office_hour_view.py [hunk 1]
```diff
@@ -24,6 +24,10 @@
 User = get_user_model()
 
 
+HOUR_MINUTE_FORMAT = "%I:%M"
+MONTH_DAY_FORMAT = "%m:%d"
```
Comment (iankigen) [inline web/impact/impact/v1/views/reserve_office_hour_view.py]: The `:` ~~notation~~ literal on the month day format is a little bit confusing. why not display full date including `%Y`?

**Comment by iankigen** [comment]
This bug also affects [cancel_office_hour_reservation](http://localhost/api/v1/cancel_office_hour_reservation/) emails. @jonkiparsky should I create a ticket for that or would you want to include that in this PR?

QA/Code review looks good

File: web/impact/impact/v1/views/reserve_office_hour_view.py (hunk 1 above)
Comment (jonkiparsky) [reply web/impact/impact/v1/views/reserve_office_hour_view.py]: The requirements for the feature specified this format. The ":" is a literal colon, not notation.

Comment (iankigen) [reply web/impact/impact/v1/views/reserve_office_hour_view.py]: Thanks for the correction. I can't find `MONTH_DAY_FORMAT = "%m:%d"` on the ticket description. This makes it hard to differentiate time from the date. example: `at 10:30 on
07:10 (America/New_York)`

File: web/impact/impact/v1/views/utils.py [hunk 2]
```diff
@@ -8,6 +8,11 @@
     UserRole,
 )
 
+HOUR_MINUTE_FORMAT = "%I:%M"
```
Comment (iankigen) [inline web/impact/impact/v1/views/utils.py]: can we include `%p` since we are using the 12-hour clock format

**Review by iankigen — APPROVED** [review]
QA/Code review looks good
- Time is displayed correctly on reservation confirmation emails/cancel office hour reservation email. ✅

**Review by PrinceDavis — CHANGES_REQUESTED** [review]
Review Outcome
- Time is displayed correctly on the mail sent out after reservation and cancellation of office hour ✅ 
- I noticed that the date also display in some sort of time format eg `07:15` instead of something like `July 15, 2020 (UTC)` :x:
- I am going to request change because of the last point above.

**Review by PrinceDavis — CHANGES_REQUESTED** [review]
last Review Round
- Time and Date appear normal on cancellation ✅ 

However this following `500` error happens when I try to reserve a session
```
host: localhost  django.request[95]: ERROR Internal Server Error: /api/v1/reserve_office_hour/
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 505, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 465, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 476, in raise_uncaught_exception
    raise exc
  File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 502, in dispatch
    response = handler(request, *args, **kwargs)
  File "/wwwroot/impact/v1/views/reserve_office_hour_view.py", line 67, in post
    self._reserve_office_hour())
  File "/wwwroot/impact/v1/views/reserve_office_hour_view.py", line 137, in _reserve_office_hour
    self._send_confirmation_emails()
  File "/wwwroot/impact/v1/views/reserve_office_hour_view.py", line 169, in _send_confirmation_emails
    True))
  File "/wwwroot/impact/v1/views/reserve_office_hour_view.py", line 185, in prepare_email_notification
    calendar_data = self.get_calendar_data(counterpart)
  File "/wwwroot/impact/v1/views/reserve_office_hour_view.py", line 256, in get_calendar_data
    timezone=tz).as_dict()
TypeError: __init__() got an unexpected keyword argument 'timezone'
```

**Review by PrinceDavis — APPROVED** [review]

**Pull request merged**